lemur/plugins/lemur_gcp/tests/test_certificates.py

```diff
@@ -54,3 +54,24 @@ def test_modify_for_gcp(original_cert_name, gcp_cert_name):
 
 def test_full_ca():
     assert certificates.full_ca(body, cert_chain) == f"{body}\n{cert_chain}"
+
+
+def test_get_self_link():
+    assert certificates.get_self_link("sandbox", "cert1") == \
+           "https://www.googleapis.com/compute/v1/projects/sandbox/global/sslCertificates/cert1"
+
+
+@pytest.mark.parametrize(
+    ("certs", "new_cert", "old_cert", "expected"),
+    [
+        (["a", "b"], "c", "b", ["a", "c"]),
+        (["a"], "c", "b", ["a", "c"]),
+        (["a", "b"], "b", "b", ["a", "b"]),
+        (["a", "b"], "c", "a", ["c", "b"]),
+        (["a", "b", "c"], "d", "e", ["a", "b", "c", "d"]),
+        (["a", "b", "c"], "c", "d", ["a", "b", "c"]),
+        (["a", "b"], "a", "b", ["a"])
```

Comment:
If both the `old_cert` and `new_cert` are present in `certs` then there still exists the possibility that the default certificate will be modified which is an unintended side-effect when performing an SNI certificate rotation. The below test case illustrates the issue:

```suggestion
        (["a", "b"], "a", "b", ["a"]),
        (["a", "b", "c"], "c", "a", ["a", "b"])
```

If the default certificate is changed when not intended, it could break clients which don't support or use SNI.

Reply:
This codepath can only be invoked when an SNI cert is being rotated. I will add an exception and update the tests if cert[0] == old cert